include/envoy/upstream/upstream.h

```diff
@@ -541,6 +541,7 @@ class PrioritySet {
   COUNTER(update_failure)                                                                          \
   COUNTER(update_no_rebuild)                                                                       \
   COUNTER(update_success)                                                                          \
+  COUNTER(upstream_buffered_request_body_buffer_overflow_total)                                    \
```

Comment:
I think maybe this should be a router filter stat vs. an upstream cluster stat? We have a few router specific stats which I think this would fit OK in vs. having it apply across all clusters? WDYT?

Reply:
I tried to keep the overflow watermark stats in the same location of the high/low watermark stats (if there were any). In this case there are other counters in this file related to high/low watermarks:
https://github.com/envoyproxy/envoy/blob/77a632737c032d0d7b5e6b7d61542287c9ceb963/include/envoy/upstream/upstream.h#L566-L570

I don't have strong objections to move it into the router, but just wondering what makes more sense.

Reply:
I could go either way on this. I'm mostly trying to avoid added cluster stats unless we really need them since they cost memory for every cluster.

If we keep them in cluster I would put this stat in the "rq" section so something like upstream_rq_buffered_body_overflow or similar.